packages/api/src/routes/connector.ts

```diff
@@ -324,35 +324,41 @@ async function getCollectionSchemaRouter(ctx: Context) {
 }
 
 async function execute(ctx: Context) {
-  const payload = plainToClass(ExecuteSqlRequest, ctx.request.body)
-  await validate(ctx, payload)
-  const user = mustGetUser(ctx)
-  const { workspaceId, connectorId, profile, sql, maxRow, questionId } = payload
-
-  const assembledSql = await translate(sql)
-
-  const manager = await getIConnectorManagerFromDB(connectorId)
-
-  const identifier = nanoid()
-
-  ctx.res.on('close', async () => {
-    await connectorService.cancelQuery(manager, identifier)
-  })
-
-  await withKeepaliveStream(ctx, async (streamResponse) => {
-    const queryResultStream = await connectorService.executeSql(
-      manager,
-      user.id,
-      workspaceId,
-      profile,
-      assembledSql,
-      identifier,
-      maxRow,
-      questionId,
-      streamHttpErrorCb(streamResponse),
-    )
-    queryResultStream.pipe(streamResponse)
-  })
+  try {
+    const payload = plainToClass(ExecuteSqlRequest, ctx.request.body)
+    await validate(ctx, payload)
+    const user = mustGetUser(ctx)
```

Comment:
If the parameter fails to be verified or users don't login, it can still use the popup message

Reply:
technically if users don't log in they would not be able to do this query (in the scenario of frontend), so maybe it is acceptable?

Reply:
make sure the logic of the api is accurate, don't consider the user's operation

Reply:
sure